src/hooks/use-http-poll.ts

```diff
@@ -0,0 +1,70 @@
+import { useCallback, useEffect, useState } from "react"
+import { useInterval } from "react-use"
+import { stateBasedArrayMerge } from "utils/state-handler"
+import { axiosInstance } from "hooks/use-http"
+
+export const DEFAULT_POLL_INTERVAL = 10_000
+export const BIGGEST_INTERVAL_NUMBER = 2 ** 31 - 1
+
+
+const createUrlWithLastUpdated = (url: string, lastUpdated: string | undefined) => (
+  lastUpdated ? `${url}?last_updated=${lastUpdated}` : url
+)
+
+interface RequestResult<T extends {}> {
+  results: T[]
+  updatedAt: string
+}
+
+// currently support only array types
+export const useHttpPoll = <T>(
+  url: string,
+  shouldMakeCall: boolean = true,
```

Comment:
Just for curiosity what is this `shouldMakeCall` flag? What could be a case where this shouldn't make the call?

Reply:
So this is my experiment with having a hook to make calls, i'm not sure yet if it went well or bad :) We don't have redux/sagas in iframes, so i thought i can create something more declarative than a lot of `useEffect`s. 

You cannot wrap hooks with `if` statements, so i put the condition to make the call into second arg, for example this:
```
  const [rooms, resetRooms] = useHttpPoll<Room>(
    `${cloudApiUrl}spaces/${spaceID || firstSpaceID}/rooms`,
    Boolean(firstSpaceID),
  )
```
will make calls when `firstSpaceID` will be truthy.

I'm actually interested if this way of managing calls (also `useHttp`) is easier or harder to reason about, so when you'll have a chance to do some coding in this repo, please let me know! :)